code/extensions/ErrorPageSubsite.php

```diff
@@ -1,37 +1,55 @@
 <?php
 
+namespace SilverStripe\Subsites\Extensions;
+
+use SilverStripe\CMS\Model\SiteTree;
+use SilverStripe\Core\Config\Config;
+use SilverStripe\ORM\DataExtension;
+use SilverStripe\ORM\DataObject;
+use SilverStripe\Subsites\Model\Subsite;
+
 class ErrorPageSubsite extends DataExtension
 {
-	/**
-	 * Alter file path to generated a static (static) error page file to handle error page template on different sub-sites
-	 *
-	 * {@see Error::get_error_filename()}
-	 *
-	 * FIXME since {@link Subsite::currentSubsite()} partly relies on Session, viewing other sub-site (including main site) between
-	 * opening ErrorPage in the CMS and publish ErrorPage causes static error page to get generated incorrectly.
-	 *
-	 * @param string $name Filename to write to
-	 * @param int $statusCode Integer error code
-	 */
-	public function updateErrorFilename(&$name, $statusCode)
+    /**
+     * Alter file path to generated a static (static) error page file to handle error page template on different sub-sites
+     *
+     * @see Error::get_filepath_for_errorcode()
+     *
+     * FIXME since {@link Subsite::currentSubsite()} partly relies on Session, viewing other sub-site (including main site) between
+     * opening ErrorPage in the CMS and publish ErrorPage causes static error page to get generated incorrectly.
+     * @param $statusCode
+     * @param null $locale
+     * @return string
+     */
+    public function alternateFilepathForErrorcode($statusCode, $locale = null)
```

Comment:
This API has been removed in 4.x. See https://github.com/silverstripe/silverstripe-errorpage#upgrading-from-silverstripe-3x